gtk/src/toga_gtk/widgets/canvas.py

```diff
@@ -81,9 +81,11 @@ def mouse_down(self, obj, event):
             pass
 
     def mouse_move(self, obj, event):
-        if event.state == Gdk.ModifierType.BUTTON1_MASK:
+        """Handles mouse movement by calling the drag and/or alternative drag
+        methods. Modifier keys have no effect."""
+        if event.state & ~Gdk.ModifierType.BUTTON3_MASK & Gdk.ModifierType.BUTTON1_MASK:
```

Comment:
The two drags aren't exclusive - if you draw with button 1 *and* button 3 pressed, you should be getting 2 notifications.
```suggestion
        if event.state & Gdk.ModifierType.BUTTON1_MASK:
```